### DISCUSSION TITLE
Can't convert Seurat 5 object to h5ad AnnData

### DISCUSSION THREAD

**Discussion by DaianeH** [satijalab/seurat#7402] [General]
Hi,

I used to work with libraries SeuratDisk or sceasy to convert between formats, but conversion is not working at the moment on a Seurat 5 object to annData.

My default assay is RNA:

sc_rna@assays
$RNA
Assay (v5) data with 29357 features for 4023 cells
Top 10 variable features:
 HES4, ISG15, UBE2J2, MRPL20, AL590822.1, PRDM16, SMIM1, DFFB, ESPN, TNFRSF9 
Layers:
 data, counts, scale.data 

But when I run sceasy::convertFormat() it cannot find the RNA assay:
Error in GetAssay.Seurat(obj, assay = assay) : 
  RNA is not an assay present in the given object.

Is there any up to date way of converting Seurat 5 objects?

Thank you in advance,

**Comment by ChristinaSteyn** [comment]
Hi there, I was having problems converting a Seurat object to an h5ad object and I found some suggestions here https://www.biostars.org/p/396351/. The SCP package eventually worked for me.

**Comment by epoisel13** [comment]
Hi,

same problem here when converting Seurat5 object using SeuratDisk. For me, the suggestions from Biostars work for v.4.x objects but somehow the Seurat 5 objects behave differently. @DaianeH have you found a solution for this?

**Comment by gvogler** [comment]
This works for me with V5 data:

```
sce_obj <- as.SingleCellExperiment(sce_obj, assay = c("RNA"))
library(zellkonverter)
writeH5AD(sce_obj, "sce_obj.h5ad", X_name = 'counts')
```

If you have several layers ("RNA", "joined"), include them in the assay parameter, e.g. ` assay = c("RNA", "joined)`.

**Reply by epoisel13** [reply]
Works for me as well, thank you for sharing this!

**Comment by kmcandre88** [comment]
Just a note if your planning on conducting RNA velocity via scvelo with the h5ad converted from the SO, I would do so manually via this tutorial's method, https://smorabit.github.io/tutorials/8_velocyto/

I had many issues trying to use various converters. I have not tried this is V5 yet but it should be similar if not the same.
